pkp/pkp-lib#11031 [stable-3_5_0] Return default user group  if  userGroup is empty

blesildaramirez opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/pkp/pkp-lib?pullRequest=11052) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/pkp/pkp-lib?pullRequest=11052) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pkp/pkp-lib?pullRequest=11052) it.</sub>

asmecher reviewed (commented):
@blesildaramirez, I think the original problem was likely in the format of the returned collection, but this change also reduces the choices a user will have in case they don't have any appropriate user groups. Is the shift from `get()` to `first()` intentional?

blesildaramirez commented:
Hi @asmecher . Yes that's intentional. It's originally using `first()` from this function `getFirstSubmitAsAuthorUserGroup`: https://github.com/pkp/pkp-lib/pull/11033/files#diff-c8cb812ae82cc024166167807c2d42818fe8c33bc61dc597835f7dbf2d7c91b2L454
We removed that function now because we moved the query execution at `getSubmitUserGroups()` function. Just that while moving, [I mistakenly used `get()`](https://github.com/pkp/pkp-lib/pull/11034#discussion_r1982665219) instead of the original `first()`.
